File: plugins/techwednesday.rb
# encoding: utf-8

require 'cinch'
require './lib/mediawiki'

class TechWednesday
    include Cinch::Plugin
    include MediaWiki

    set :help, <<-EOF
!tw
  Affiche l'ordre du jour du prochain Tech Wednesday
!tw <votre point>
  Ajoute <votre point> à l'ordre du jour du prochain TechWednesday
EOF

    match /tw$/, :method => :tw
    match /tw (.+)/, :method => :tw
    def tw msg, point=''
        point.strip!
        msg.reply "Un instant..."

        next_tw_page = next_tw login(config[:wiki_url], config[:username], config[:password])

        unless next_tw_page
            msg.reply "Prochain TechWednesday introuvable..."
            return
        end

        date = find_tw_date next_tw_page
        tw_str = date.strftime "TechWednesday (%d/%m/%Y)"

        if point.empty?
            msg.reply "Prochain #{tw_str}:"
            next_tw_oj(next_tw_page).each do |point|
                msg.reply " - #{point}"
            end
        else
            point = "#{point} (#{msg.user})"
            next_tw_add_point next_tw_page, point
            msg.reply "« #{point} » ajouté à l'ordre du jour du prochain #{tw_str}"
        end

        msg.reply("#{next_tw_page.uri}")
    end
end

File: plugins/techwednesday.py
from ircbot.plugin import BotPlugin

from .helpers import mkurl, private_api, public_api


class TechWednesday(BotPlugin):
    @BotPlugin.command(r"\!tw +([^ ].+)")
    async def add_to_next_tw(self, msg):
        """Ajoute un point à l'ordre du jour de la prochaine réunion"""
        await private_api(
            "/events/add_point_to_next_meeting",
            {"point": msg.args[0] + " (" + msg.user + ")"},
        )
        msg.reply(
            'Point "' + self.bot.text.bold(msg.args[0]) + "\" ajouté à l'ordre du jour",
            hilight=True,
        )
        self.bot.log.info('Add "' + msg.args[0] + '" to next tw by ' + msg.user)

    @BotPlugin.command(r"\!tw")
    async def next_tw(self, msg):
        """Affiche l'ordre du jour de la prochaine réunion"""
        try:
            next_meeting = await public_api("/events/next_meeting")
            when = self.bot.naturaltime(next_meeting["event"]["start"])
            msg.reply(self.bot.text.bold("Prochaine réunion " + when))
            for line in next_meeting["OJ"].split("\n"):
                line = line.strip()
                if line:
                    msg.reply(line)
            msg.reply(
                self.bot.text.blue(mkurl("events/" + str(next_meeting["event"]["id"])))
            )
        except:
            msg.reply("Pas de prochain tw trouvé")
